util/reggen/register.py

```diff
@@ -295,18 +295,24 @@ def from_raw(reg_width: int,
                                 'fields for {} register'.format(name))
         if not raw_fields:
             raise ValueError('Register {} has no fields.'.format(name))
-        fields = [Field.from_raw(name,
-                                 idx,
-                                 len(raw_fields),
-                                 swaccess,
-                                 hwaccess,
-                                 resval,
-                                 reg_width,
-                                 params,
-                                 hwext,
-                                 shadowed,
-                                 rf)
-                  for idx, rf in enumerate(raw_fields)]
+
+        fields = []
+        used_bitmask = 0
+        for idx, rf in enumerate(raw_fields):
+            fields.append(Field.from_raw(name,
+                                         idx,
+                                         len(raw_fields),
+                                         swaccess,
+                                         hwaccess,
+                                         resval,
+                                         reg_width,
+                                         params,
+                                         hwext,
+                                         shadowed,
+                                         rf,
+                                         used_bitmask))
+
+            used_bitmask |= fields[idx].bits.bitmask()
```

Comment:
I'd prefer not to add another argument to `Field`'s constructor. The point is that if I have overlapping bits, the fields themselves are perfectly fine. The problem is that they shouldn't both live in the same register! So I think it would be better to do something like this:
```python
        fields = []
        used_bits = 0
        for idx, raw_field in enumerate(raw_fields):
            field = Field.from_raw(name,
                                   idx,
                                   len(raw_fields),
                                   swaccess,
                                   hwaccess,
                                   resval,
                                   reg_width,
                                   params,
                                   hwext,
                                   shadowed,
                                   raw_field)
            overlap_bits = used_bits & field.bits.bitmask()
            if overlap_bits:
                raise ValueError(f'Field {field.name} uses bits '
                                 f'{overlap_bits:#x} that appear in other '
                                 f'fields.')
            used_bits |= field.bits.bitmask()
            fields.append(field)
```